client/src/components/AuthModal.jsx

```diff
@@ -0,0 +1,282 @@
+    display: "inline",
+    color: "#DF1B1B",
+    fontWeight: "bold",
+    cursor: "pointer",
+  },
+}));
+
+const AuthModal = ({ isAuthPage }) => {
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [signedIn, setSignedIn] = useState(true);
+  const [isFirst, setIsFirst] = useState(true);
+
+  const history = useHistory();
+
+  useEffect(async () => {
+    setIsFirst(false);
+    let results = await fetch("/user/getCurrentUser");
+
+    results = await results.json();
+
+    if (results.user === undefined) {
+      setSignedIn(false);
+    } else {
+      setSignedIn(true);
+      history.push("/dashboard");
+    }
+  }, []);
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+
+    if (password.length < 7) {
+      alert("Password must be greater than 6");
+      return;
+    }
+
+    if (name.trim().length === 0 && pathname === "/signup") {
+      alert("Name must not be just spaces");
+      return;
+    }
+
+    try {
+      //email is handled by the TextField
+      if (pathname === "/signup") {
+        let result = await fetch("/user/register", {
+          method: "POST",
+          headers: { "Content-type": "application/json" },
+          body: JSON.stringify({
+            name,
+            email,
+            password,
+          }),
+        });
+
+        result = await result.json();
+
+        setSignedIn(
+          result.user !== undefined && result.user.id !== undefined
+            ? true
+            : false
+        );
+      } else {
+        let result = await fetch("/user/login", {
+          method: "POST",
+          headers: { "Content-type": "application/json" },
+          body: JSON.stringify({
+            email,
+            password,
+          }),
+        });
+
+        result = await result.json();
+
+        setSignedIn(
+          result.user !== undefined && result.user.id !== undefined
+            ? true
+            : false
+        );
+      }
+      history.push("/dashboard");
+    } catch (error) {
+      alert("Something went wrong");
+      console.error(error);
+    }
+  };
+
+  const classes = useStyles();
+  const { pathname } = window.location;
+
+  return (
+    <>
+      {signedIn  ? (
+        <></>
+      ) : (
+        <Modal open={!signedIn}  hideBackdrop={isAuthPage}>
+          <Paper id="auth-paper" className={classes.paper} elevation={3}>
+            <Typography variant="h4" className={classes.h4}>
+              {pathname === "/signup" ? "Sign Up" : "Sign In"}
+            </Typography>
+
+            <form onSubmit={submitHandler} style={{ display: "block" }}>
+              <Grid
+                container
+                direction="column"
+                justify="center"
+                alignItems="center"
+              >
+                {pathname === "/signup" && (
+                  <Grid item md={"auto"}>
+                    <Box className={classes.box}>
```

Comment:
You could probably get rid of this `<Box>`, and move the `className` to the `<Grid>` above.